propagation.go

```diff
@@ -259,6 +267,29 @@ func (p *binaryPropagator) Extract(abstractCarrier interface{}) (SpanContext, er
 	return ctx, nil
 }
 
+// Converts a comma separated key value pair list into a map
+// e.g. key1=value1, key2=value2, key3 = value3
+// is converted to map[string]string { "key1" : "value1",
+//                                     "key2" : "value2",
+//                                     "key3" : "value3" }
+func parseCommaSeparatedMap(value string) map[string]string {
+	baggage := make(map[string]string)
+	kvpairs := strings.Split(value, ",")
+	for _, kvpair := range kvpairs {
+		s, err := url.QueryUnescape(strings.TrimSpace(kvpair))
+		if err != nil {
+			log.Printf("Unable to unescape %s", kvpair)
+		}
```

Comment:
I am not saying remove logging, but currently you're proceeding with an empty `s`. So either need to add `continue` or `s = kvpair`